src/coreclr/jit/lowerxarch.cpp

```diff
@@ -8202,18 +8202,20 @@ bool Lowering::IsContainableHWIntrinsicOp(GenTreeHWIntrinsic* parentNode, GenTre
         case NI_AVX2_BroadcastScalarToVector256:
         case NI_AVX512F_BroadcastScalarToVector512:
         {
-            var_types baseType = hwintrinsic->GetSimdBaseType();
-            if (varTypeIsSmall(baseType))
+            var_types parentBaseType = parentNode->GetSimdBaseType();
+            var_types childBaseType  = hwintrinsic->GetSimdBaseType();
+
+            if (varTypeIsSmall(parentBaseType) || varTypeIsSmall(childBaseType))
```

Comment:
Is this sufficient? For example, doesn't
```csharp
[MethodImpl(MethodImplOptions.NoInlining)]
public static Vector256<long> Test(Vector256<long> x, ref int y)
{
    return Avx2.Add(x + x, Vector256.Create(y).AsInt64());
}
```
have the same problem and still pass these checks?

Reply:
Yes, good point. Will do a check that the sizes match here.

There's a "todo" in that it is safe for some constants, but that's a different path and is already handled.

Reply:
Fixed.